tests/test_cases/test_cocotb/test_edge_triggers.py

```diff
@@ -30,16 +29,14 @@ def do_single_edge_check(dut, level):
     """Do test for rising edge"""
     old_value = dut.clk.value.integer
     dut._log.info("Value of %s is %d" % (dut.clk._path, old_value))
-    if old_value is level:
-        raise TestError("%s not to %d start with" % (dut.clk._path, not level))
+    assert old_value != level, "%s not to %d start with" % (dut.clk._path, not level)
     if level == 1:
         yield RisingEdge(dut.clk)
     else:
         yield FallingEdge(dut.clk)
     new_value = dut.clk.value.integer
     dut._log.info("Value of %s is %d" % (dut.clk._path, new_value))
-    if new_value is not level:
-        raise TestError("%s not %d at end" % (dut.clk._path, level))
+    assert new_value == level, "%s not %d at end" % (dut.clk._path, level)
```

Comment:
The differentiation between ``TestError`` and ``TestFailure`` is now lost but was it ever useful here?

Reply:
Here? Nope. Elsewhere? Yes and no, the test could be set to `expect_error`, but the user should never be throwing `TestError` so that's a bad justification. I think `TestError` can be deprecated at this point? It was only ever used for internal test handling, but that's all been refactored out IIRC.

Reply:
I opened a new issue about the deprecation: #2174